modules/gdscript/gdscript_cache.cpp

```diff
@@ -200,17 +206,21 @@ Ref<GDScript> GDScriptCache::get_full_script(const String &p_path, Error &r_erro
 	}
 
 	if (script.is_null()) {
-		script = get_shallow_script(p_path);
-		ERR_FAIL_COND_V(script.is_null(), Ref<GDScript>());
+		script = get_shallow_script(p_path, r_error);
+		if (r_error) {
+			return script;
+		}
```

Comment:
This is not equivalent to the previous logic. Now it seems that if the parsing fails, the script will not be cached by `shallow_gdscript_cache` (the previous logic is only an invalid script will report an error, this will hardly appear, may it appear in multi-threading?)

Reply:
you're right, i completely missed that, and just after #68914 was merged too

Reply:
ok I made #68927 to fix this